src/Merchants.php

```diff
@@ -142,15 +142,36 @@ public static function update_or_create_merchant() {
 			'merchant_name'    => $merchant_name,
 		);
 
-		// The response only contains the merchant id.
-		$response = API\Base::update_or_create_merchant( $args );
+		$cache_key = PINTEREST_FOR_WOOCOMMERCE_PREFIX . '_request_' . md5( wp_json_encode( $args ) );
+		$cache     = get_transient( $cache_key );
+
+		if ( false !== $cache ) {
+			return $cache;
+		}
+
+		try {
+			// The response only contains the merchant id.
+			$response = Base::update_or_create_merchant( $args );
+		} catch ( Throwable $th ) {
+			$delay = Pinterest_For_Woocommerce()::get_data( 'create_merchant_delay' ) ?? 10 * MINUTE_IN_SECONDS;
+
+			set_transient( $cache_key, $response, $delay );
+
+			// Double the delay.
+			Pinterest_For_Woocommerce()::save_data( 'create_merchant_delay', $delay * 2 );
```

Comment:
To prevent crazy values we should use max allowed value.
```suggestion
			Pinterest_For_Woocommerce()::save_data( 'create_merchant_delay', max( $delay * 2, HOUR_IN_SECONDS * 6 );
```